06_puppy/lexual/puppy.go

```diff
@@ -0,0 +1,96 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"sync"
+)
+
+var out io.Writer = os.Stdout
+
+type Puppy struct {
+	ID     uint32
+	Breed  string
+	Colour string
+	Value  int
+}
+
+// Storer defines standard CRUD operations for Puppys
+type Storer interface {
+	CreatePuppy(*Puppy)
+	ReadPuppy(id uint32) *Puppy
+	UpdatePuppy(id uint32, puppy *Puppy)
+	DeletePuppy(id uint32) bool
+}
+
+// MapStore ////
+
+type MapStore map[uint32]Puppy
+
+// CreatePuppy stores a copy of Puppy p
+func (store *MapStore) CreatePuppy(p *Puppy) {
+	(*store)[p.ID] = Puppy{p.ID, p.Breed, p.Colour, p.Value}
```

Comment:
No need to verbosely create a new puppy. Just `(*store)[p.ID] = *p` will do the job - since the map holds a value not a pointer, it will copy on the way in.